merge-all.rb

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env ruby
+
+# Usage:
+# "REPOS_PATH=infrastructure GH_ACCESS_TOKEN=abc123 ./merge-all.rb"
+
+BRANCH_NAME = 'update-dependencies'
+
+def ruby_repos_file
+  File.open("#{ENV['REPOS_PATH']}/ruby")
+end
+
+def repos
+  [ruby_repos_file].map(&:to_a).flatten.map(&:strip).sort
+end
+
+# @return [Array<Hash>] the update PR
+def find_prs(client, repos)
+  repo_content = repos.map do |repo|
+    prs = client.pull_requests(repo, head: "sul-dlss:#{BRANCH_NAME}")
+
+    next if prs.empty?
+    statuses = client.combined_status(repo, prs.first.head.sha)
+
+    # GitHub API marks PRs with 0 statuses as "pending", we cast that to success
+    status = statuses.total_count == 0 ? 'success' : statuses.state
+    { repo: repo, number: prs.first.number, url: prs.first.url, status: status }
+  end.compact
+end
+
+def access_token
+  ENV['GH_ACCESS_TOKEN']
+end
+
+require 'byebug'
+require 'octokit'
+client = Octokit::Client.new(access_token: access_token)
+pr_list = find_prs(client, repos)
+
+
+if pr_list.empty?
+  puts "No PRs were found"
+  exit
+end
+
+unless pr_list.all? { |pr| pr[:status] == 'success' }
```

Comment:
The script only merges any PRs if all are successful? Curious why we might not want to filter out the busted builds (with useful information spat out) and proceed with the ones that are :green_heart:

Reply:
I guess this is just being extra cautious for the time being. We wouldn't want to push an dor-services-client update if it fails in one place as the API might not be compatible.

Reply:
ok, I'm fine with that